ecosystem/typescript/sdk/src/transaction_builder/transaction_vector.test.ts

```diff
@@ -0,0 +1,228 @@
+  VECTOR_FILES_ROOT_DIR,
+  'aptos_api__tests__transaction_vector_test__test_script_payload.json',
+);
+
+const MODULE_VECTOR = path.join(
+  VECTOR_FILES_ROOT_DIR,
+  'aptos_api__tests__transaction_vector_test__test_module_payload.json',
+);
+
+function parseTypeTag(typeTag: any): TypeTag {
+  if (typeTag.vector) {
+    return new TypeTagVector(parseTypeTag(typeTag.vector));
+  }
+
+  if (typeTag.struct) {
+    const {
+      address,
+      module,
+      name,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      type_args,
+    }: {
+      address: string;
+      module: string;
+      name: string;
+      type_args: any[];
+    } = typeTag.struct;
+
+    const typeArgs = type_args.map((arg) => parseTypeTag(arg));
+    const structTag = new StructTag(
+      AccountAddress.fromHex(address),
+      new Identifier(module),
+      new Identifier(name),
+      typeArgs,
+    );
+
+    return new TypeTagStruct(structTag);
+  }
+
+  switch (typeTag) {
+    case 'bool':
+      return new TypeTagBool();
+    case 'u8':
+      return new TypeTagU8();
+    case 'u64':
+      return new TypeTagU64();
+    case 'u128':
+      return new TypeTagU128();
+    case 'address':
+      return new TypeTagAddress();
+    case 'signer':
+      return new TypeTagSigner();
+    default:
+      throw new Error('Unknown type tag');
+  }
+}
+
+function parseTransactionArgument(arg: any): TransactionArgument {
+  const argHasOwnProperty = (propertyName: string) => Object.prototype.hasOwnProperty.call(arg, propertyName);
+  if (argHasOwnProperty('U8')) {
+    // arg.U8 is a number
+    return new TransactionArgumentU8(arg.U8);
+  }
+
+  if (argHasOwnProperty('U64')) {
+    // arg.U64 is a string literal
+    return new TransactionArgumentU64(BigInt(arg.U64));
+  }
+
+  if (argHasOwnProperty('U128')) {
+    // arg.U128 is a string literal
+    return new TransactionArgumentU128(BigInt(arg.U128));
+  }
+
+  if (argHasOwnProperty('Address')) {
+    // arg.Address is a hex string
+    return new TransactionArgumentAddress(AccountAddress.fromHex(arg.Address));
+  }
+
+  if (argHasOwnProperty('U8Vector')) {
+    // arg.U8Vector is a hex string
+    return new TransactionArgumentU8Vector(new HexString(arg.U8Vector).toUint8Array());
+  }
+
+  if (argHasOwnProperty('Bool')) {
+    return new TransactionArgumentBool(arg.Bool);
+  }
+
+  throw new Error('Invalid Transaction Argument');
+}
+
+function sign(rawTxn: RawTransaction, privateKey: string): string {
+  const privateKeyBytes = new HexString(privateKey).toUint8Array();
+  const signingKey = Nacl.sign.keyPair.fromSeed(privateKeyBytes.slice(0, 32));
+  const { publicKey } = signingKey;
+
+  const txnBuilder = new TransactionBuilderEd25519(
+    (signingMessage) => new Ed25519Signature(Nacl.sign(signingMessage, signingKey.secretKey).slice(0, 64)),
+    publicKey,
+  );
+
+  return Buffer.from(txnBuilder.sign(rawTxn)).toString('hex');
+}
+
+function doVerification(
+  raw_txn: Record<string, any>,
```

Comment:
might be good to actually type it out since these contain a lot of fixed pieces?